src/components/Workshops/index.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState, useMemo, useEffect } from "react";
+import dayjs from "dayjs";
+import {
+  Workshop,
+  WorkshopLevel,
+  WorkshopSortType,
+  WorkshopTag,
+} from "../../types";
+import { WorkshopTagButton } from "../WorkshopTagButton";
+import { SortButton } from "./components/SortButton";
+import { sortEvents } from "./helpers";
+import { WorkshopItem } from "../WorkshopItem";
+
+enum PriceFilter {
+  Free,
+  Paid,
+}
+
+type Props = {
+  events: Workshop[];
+  tags: WorkshopTag[];
+  levels: WorkshopLevel[];
+};
+
+export const Workshops = ({ events, tags, levels }: Props) => {
+  // aktualne sortowanie i aktualnie wybrane wydarzenia
+  const [sortType, setSortType] = useState<WorkshopSortType>(
+    WorkshopSortType.ByDate
+  );
+  const [currentTags, setCurrentTags] = useState<WorkshopTag[]>([]);
+  const [currentLevels, setCurrentLevels] = useState<WorkshopLevel[]>([]);
+  const [currentPriceFilters, setCurrentPriceFilters] = useState<PriceFilter[]>(
+    []
+  );
+
+  const toggleTag = (tag: WorkshopTag) =>
+    setCurrentTags(
+      currentTags.includes(tag)
+        ? currentTags.filter((current) => current.name !== tag.name)
+        : [...currentTags, tag]
+    );
+  const toggleLevel = (level: WorkshopLevel) =>
+    setCurrentLevels(
+      currentLevels.includes(level)
+        ? currentLevels.filter((current) => current.name !== level.name)
+        : [...currentLevels, level]
+    );
+
+  const togglePrice = (filter: PriceFilter) =>
+    setCurrentPriceFilters(
+      currentPriceFilters.includes(filter)
+        ? currentPriceFilters.filter((current) => current !== filter)
+        : [...currentPriceFilters, filter]
+    );
+
+  const filteredEvents = useMemo(
+    () =>
+      events.filter(() => {
+        // tu powinna przejść pętla która na podstawie leveli, cen i tagów odfiltruje co trzeba :)
+        // zmienne z nimi juz masz tylko trzeba to opisac
+        // i dodać do aktualizowanych pól poniej
+        return true;
+      }),
+    [events]
+  );
+
+  const oldEvents = useMemo(
+    () =>
+      sortEvents(
+        filteredEvents.filter(
+          (event) => dayjs(event.startDate).unix() < dayjs().unix()
+        ),
+        sortType,
+        true
+      ),
+    [filteredEvents, sortType]
+  );
+
+  const onGoingEvents = useMemo(
+    () =>
+      sortEvents(
+        filteredEvents.filter(
+          (event) => dayjs(event.startDate).unix() > dayjs().unix()
+        ),
+        sortType,
+        false
+      ),
+    [filteredEvents, sortType]
+  );
```

Comment:
Most of this logic is redundant and could be moved into a helper or even custom hook.
`const { oldEvents, ongoingEvents } = sortEvents(filteredEvents, sortType)`

Reply:
I could do a PR with that.

Reply:
feel free to improve - i just made a draft from Tomek's code - it's not a real PR :P